backend/e4h-services/ingestion-service/app/api/endpoints/file_ingestion.py

```
@@ -989,4 +990,107 @@ async def check_duplicate_tickets(
+        df.to_excel(output_temp_file.name, sheet_name=incidents_sheet_name, index=False)
+
+        return FileResponse(
+            path=output_temp_file.name,
+            filename=output_filename,
+            media_type="application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+        )
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Failed to process incident updates: {str(e)}"
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)